arangod/RocksDBEngine/RocksDBOptionFeature.cpp

```diff
@@ -1526,6 +1527,37 @@ limited number of edge collections/shards/indexes.)");
   can open. Thus the option should only be enabled on deployments with a
   limited number of edge collections/shards/indexes.)");
 
+  options
+      ->addOption("--rocksdb.partition-files-for-vector-index",
+                  "If enabled, the index data for different vector "
+                  "indexes will end up in different .sst files.",
+                  new BooleanParameter(&_partitionFilesForVectorIndexCf),
+                  arangodb::options::makeFlags(
+                      arangodb::options::Flags::Uncommon,
+                      arangodb::options::Flags::Experimental,
+                      arangodb::options::Flags::DefaultNoComponents,
+                      arangodb::options::Flags::OnDBServer,
+                      arangodb::options::Flags::OnSingle))
+      .setIntroducedIn(31204)
+      .setLongDescription(R"(Enabling this option will make RocksDB's
+  compaction write the index data for different vector
+  indexes (also indexes from different collections/shards) into different
+  .sst files. Otherwise the index data from different
+  collections/shards/indexes can be mixed and written into the same .sst files.
+
+  Enabling this option usually has the benefit of making the RocksDB
+  compaction more efficient when a lot of different collections/shards/indexes
+  are written to in parallel.
+  The disadvantage of enabling this option is that there can be more .sst
+  files than when the option is turned off, and the disk space used by
+  these .sst files can be higher than if there are fewer .sst files (this
+  is because there is some per-.sst file overhead).
+  In particular on deployments with many collections/shards/indexes
+  this can lead to a very high number of .sst files, with the potential
+  of outgrowing the maximum number of file descriptors the ArangoDB process
+  can open. Thus the option should only be enabled on deployments with a
+  limited number of edge collections/shards/indexes.)");
```

Comment:
```suggestion
      .setLongDescription(R"(Enabling this option makes RocksDB's
compaction write the index data for different vector
indexes (also indexes from different collections/shards) into different
.sst files. Otherwise, the index data from different
collections/shards/indexes can be mixed and written into the same .sst files.

Enabling this option usually has the benefit of making the RocksDB
compaction more efficient when a lot of different collections/shards/indexes
are written to in parallel.
The disadvantage of enabling this option is that there can be more .sst
files than when the option is disabled, and the disk space used by
these .sst files can be higher than if there are fewer .sst files
because there is some overhead per .sst file.
For deployments with many collections/shards/indexes in particular,
this can lead to a very high number of .sst files, with the potential
of outgrowing the maximum number of file descriptors the ArangoDB process
can open. The option should thus only be enabled for deployments with a
limited number of edge collections/shards/indexes.)");
```